src/main/java/net/sf/jabref/gui/importer/fetcher/DOAJFetcher.java

```diff
@@ -43,31 +44,26 @@ public boolean processQuery(String query, ImportInspector inspector, OutputPrint
             HttpResponse<JsonNode> jsonResponse;
             jsonResponse = Unirest.get(SEARCH_URL + query + "?pageSize=1").header("accept", "application/json").asJson();
             JSONObject jo = jsonResponse.getBody().getObject();
-            int hits = jo.getInt("total");
-            int numberToFetch = 0;
-            if (hits > 0) {
-                if (hits > MAX_PER_PAGE) {
-                    while (true) {
-                        String strCount = JOptionPane
-                                .showInputDialog(
-                                        Localization.lang("References found") + ": " + hits + "  "
-                                                + Localization.lang("Number of references to fetch?"),
-                                        Integer.toString(hits));
+            int numberToFetch = jo.getInt("total");
+            if (numberToFetch > 0) {
+                if (numberToFetch > MAX_PER_PAGE) {
+                    boolean numberEntered = false;
+                    do {
+                        String strCount = JOptionPane.showInputDialog(Localization.lang("References found") +": " + numberToFetch + " " +
```

Comment:
It would be best to bind the number to the translation, but then it has to be re-translated b/c the number isn't always at the same spot in every language.